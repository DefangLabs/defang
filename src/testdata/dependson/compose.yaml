services:
  service1:
    image: alpine
    depends_on:
      - service2
      - service3
  service2:
    image: alpine
    depends_on:
      - service3
  service3:
    image: alpine
    depends_on:
      - service3
