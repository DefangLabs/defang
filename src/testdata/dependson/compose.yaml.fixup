{
  "service1": {
    "command": null,
    "depends_on": {
      "service2": {
        "condition": "service_started",
        "required": true
      },
      "service3": {
        "condition": "service_started",
        "required": true
      }
    },
    "deploy": {
      "replicas": 1,
      "resources": {},
      "placement": {}
    },
    "entrypoint": null,
    "image": "alpine",
    "networks": {
      "default": null
    }
  },
  "service2": {
    "command": null,
    "depends_on": {
      "service3": {
        "condition": "service_started",
        "required": true
      }
    },
    "deploy": {
      "replicas": 1,
      "resources": {},
      "placement": {}
    },
    "entrypoint": null,
    "image": "alpine",
    "networks": {
      "default": null
    }
  },
  "service3": {
    "command": null,
    "depends_on": {
      "service3": {
        "condition": "service_started",
        "required": true
      }
    },
    "deploy": {
      "replicas": 1,
      "resources": {},
      "placement": {}
    },
    "entrypoint": null,
    "image": "alpine",
    "networks": {
      "default": null
    }
  }
}