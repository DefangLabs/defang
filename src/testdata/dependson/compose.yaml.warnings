 ! can only have more than one replica for PRO tier and PRODUCTION deployment mode. Will set replicas to 1 for service(s) [service1 service2 service3]
 ! service "service1": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "service2": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "service3": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors