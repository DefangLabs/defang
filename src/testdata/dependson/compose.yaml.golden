name: dependson
services:
  service1:
    depends_on:
      service2:
        condition: service_started
        required: true
      service3:
        condition: service_started
        required: true
    image: alpine
    networks:
      default: null
  service2:
    depends_on:
      service3:
        condition: service_started
        required: true
    image: alpine
    networks:
      default: null
  service3:
    depends_on:
      service3:
        condition: service_started
        required: true
    image: alpine
    networks:
      default: null
networks:
  default:
    name: dependson_default
